llvm/lib/Analysis/BasicAliasAnalysis.cpp

```diff
@@ -997,6 +998,19 @@ ModRefInfo BasicAAResult::getModRefInfo(const CallBase *Call,
   }
 
   ModRefInfo Result = ArgMR | OtherMR;
+
+  // Refine writes to errno memory. We can safely exclude the errno location if
+  // the given memory location is an identified function-local object, the size
+  // of the memory access is larger than the integer size, or if TBAA proves it
+  // does not alias errno.
+  if ((ErrnoMR | Result) != Result) {
+    if (ErrnoMR == ModRefInfo::Mod && !isIdentifiedFunctionLocal(Object) &&
```

Comment:
Why is this limited to `ErrnoMR == ModRefInfo::Mod`? Shouldn't this logic apply to both mod and ref of errno?